.github/workflows/check-code.yml

```diff
@@ -13,192 +13,288 @@ concurrency:
+  #   needs: cargo-clippy
+
+  #   strategy:
+  #     matrix:
+  #       features:
+  #         -
+  #         - --all-features
+  #     fail-fast: false
+
+  #   steps:
+  #     - name: Checkout repository
+  #       uses: actions/checkout@v4
+
+  #     - name: Free Disk Space
+  #       uses: jlumbroso/free-disk-space@main
+  #       with:
+  #         tool-cache: true
+
+  #     - name: Set up Cargo cache
+  #       uses: actions/cache@v4
+  #       with:
+  #         path: |
+  #           ~/.cargo/bin/
+  #           ~/.cargo/registry/index/
+  #           ~/.cargo/registry/cache/
+  #           ~/.cargo/git/db/
+  #         key: ${{ github.job }}-${{ github.ref }}-${{ hashFiles('**/Cargo.lock') }}
+
+  #     - name: Run `cargo doc`
+  #       run: |
+  #         docker run --rm \
+  #           -v "${GITHUB_WORKSPACE}:/workspace" \
+  #           -v "${HOME}/.cargo:/root/.cargo" \
+  #           -w /workspace \
+  #           -e RUSTDOCFLAGS='-D warnings' \
+  #           -e SKIP_WASM_BUILD=1 \
+  #           paritytech/ci-unified:bullseye-1.74.0 \
+  #           bash -c "cargo doc --no-deps --locked ${{ matrix.features }}"
+
+  # try-runtime:
+  #   name: Run try-runtime
+  #   runs-on: ubuntu-latest
+  #   needs: cargo-clippy
+  #   env:
+  #     TRY_RUNTIME_CLI_VERSION_TAG: v0.7.0
+  #   container:
+  #     image: paritytech/ci-unified:bullseye-1.74.0
+
+  #   strategy:
+  #     matrix:
+  #       runtime:
+  #         - peregrine
+  #         - spiritnet
+  #     fail-fast: false
+
+  #   steps:
+  #     - name: Checkout repository
+  #       uses: actions/checkout@v4
+
+  #     - name: Free Disk Space
+  #       uses: jlumbroso/free-disk-space@main
+  #       with:
+  #         tool-cache: true
+
+  #     - name: Install try-runtime
+  #       run: |
+  #         curl -sL https://github.com/paritytech/try-runtime-cli/releases/download/${{ env.TRY_RUNTIME_CLI_VERSION_TAG }}/try-runtime-x86_64-unknown-linux-musl -o try-runtime
+  #         chmod +x ./try-runtime
+  #         ./try-runtime --version
+
+  #     - name: Set up Cargo cache
+  #       uses: actions/cache@v4
+  #       with:
+  #         path: |
+  #           ~/.cargo/bin/
+  #           ~/.cargo/registry/index/
+  #           ~/.cargo/registry/cache/
+  #           ~/.cargo/git/db/
+  #         key: ${{ github.job }}-${{ github.ref }}-${{ hashFiles('**/Cargo.lock') }}
+
+  #     - name: Build runtime
+  #       run: cargo build --release --locked -p ${{ matrix.runtime }}-runtime --features try-runtime
+
+  #     - name: Run `try-runtime`
+  #       run: |
+  #         ./try-runtime \
+  #         --runtime=./target/release/wbuild/${{ matrix.runtime }}-runtime/${{ matrix.runtime }}_runtime.compact.compressed.wasm \
+  #         on-runtime-upgrade \
+  #           --disable-spec-version-check \
+  #           --checks=all \
+  #         live \
+  #           --uri=wss://${{ matrix.runtime }}.kilt.io
+
+  runtime-benchmarks:
+    name: Test benchmarks
```

Comment:
I think it is not necessary to run the benchmarks on every PR. Only updating polkadot dependencies can cause issues in the benchmarking logic. To speed up the CI maybe we should trigger them, maybe manually? I have also concerns to automate that based on the PR title.

Reply:
It was for testing. I changed it to run only for `push` events, so on `master`, `develop`, and for tags. I think it does not hurt to have it there. Any other solution would most likely end up being too complicated, unless you have a solution in mind.

Reply:
I see. I think we should maybe change that to `pull_request`?
According to the documentation, the pipeline will be triggered when a PR is created or updated. It would be a minor improvement, allowing someone to test things without using up CI resources. But we can also leave it like that. No strong opinion

Reply:
Exactly, we don't want to run in on every PR, right? Yet, we should be able to catch any breaking changes before we start the release process, that's why I thought having it run on pushes onto our two main branches is a good idea. Plus, if the benchmark command breaks, we should avoid push those Docker images out, that's why it's part of the same workflow.

Reply:
I agree. Should be fine.